flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -240,7 +240,9 @@ public Throwable getFailureCause() {
 
     @Override
     public void releaseAllResources() throws IOException {
-        subpartitionView.releaseAllResources();
+        if (subpartitionView != null) {
+            subpartitionView.releaseAllResources();
+        }
```

Comment:
? I'm confused by this change and the commit message. How is it related to FLINK-9057?

Reply:
Before FLINK-9057, we called notifyReaderCreated before requestSubpartitionView but it leaded to NPE in releaseAllResources so during the fixing FLINK-9057 the order was changed and notifyReaderCreated started to call after requestSubpartitionView which fixed the problem with NPE. Right now, I returned the original order(notifyReaderCreated is before requestSubpartitionView) so because of it, I should somehow fix the NPE.
But I agree that perhaps, it doesn't make sense to use FLINK-9057 in the commit. Maybe I should fix the NPE under the current ticket.